guides/dsl.md

```diff
@@ -0,0 +1,289 @@
+
+```javascript
+program.bind("Invite classmates to my party", ({find, record}) => {
+  // search for records tagged "friend"
+  let student = find("student");
+  return [
+    // bind a single record tagged "guest-list", with every student as a guest
+    record("guest-list").add({guest: student})
+  ];
+});
+```
+
+### remove()
+
+Removes attribute/value pairs from a record. This is equivalent to the Eve `-=` operator. Remove should only be used in the context of a `commit` block. For example:
+
+```javascript
+program.commit("blacklisted people cannot come to the party", ({find, record}) => {
+  let blacklisted = find("blacklisted");
+  let guests = find("guest-list")
+  return [
+    // remove anyone blacklisted from the guest list
+    guests.remove({guest: blacklisted})
+  ];
+});
+```
+
+You can also use `remove()` to completely delete records or attributes on a record. This is equivalent to `:= none` in Eve. For example:
+
+```javascript
+   // completely remove the target record
+   record.remove() 
+   // remove the attribute on the target record
+   record.remove("attribute")
+```
+
+The DSL does not contain an equivalent for the Eve set operator `:=`, but you can mimic its behavior by chaining the `add()` and `remove()` operators. For example:
+
+```javascript
+program.bind("Reassign Artemis' teacher", ({find, record}) => {
+  let artemis = find("student", {name: "Artemis"});
+  let teacher = find("teacher", {name: "Smith"});
+  return [
+    artemis.remove("teacher").add("teacher", teacher)
+  ];
+});
+```
+
+In summary:
+
+- Add a value: `person.add("salary", 10)` <-> `person.salary += 10`
+- Remove a value: `person.remove("salary, 10)` <-> `person.salary -= 10`
+- Set a value: `person.remove("salary").add("salary", 10)` <-> `person.salary := 10`
+- Remove an attribute `person.remove("salary")` <-> `person.salary := none`
+- Remove a record `person.remove()` <-> `person := none`
+
+## Sub-Blocks
+
+In the DSL `not()`, `union()`, and `choose()` are sub-blocks, which have their own body. Let's look at each of these.
+
+### not()
+
+The `not()` sub block works similarly to `not()` from the Eve syntax; it performs an anti-join on the records inside and outside of the `not()`. To use not, include it in the function list at the beginning of the block.
+
+```javascript
+program.bind("Tag students without any citations.", ({find, record, not}) => {
+    let students = find("student");
+    not(() => {
+        find("student", {citation});
+    });
+    return [
+        students.add("tag","good-standing")
+    ];
+})
+```
+
+### choose()
+
+Choose() and union() expressions are behind the mechanics of the if expression in the Eve syntax. In the DSL, we expose these directly. First, `choose()` takes a body of branches, each one comprised of a condition and a return value. The value of the choose corresponds to the first matching branch. 
```

Comment:
This is a bit confusingly worded. I'd describe it more like "choose takes a list of subblocks which contain any valid eve code to join, filter, mash, or compute their results. Each is executed in order, and the first to succeed is taken as the choose's return value. If no subblock succeeds, the row fails."